### PR TITLE
Typed Contracts

### PR DESCRIPTION
**Pull request by ditchfieldcaleb**

### CODE REVIEW

File: types/types.ts [hunk 1]
```diff
@@ -98,69 +137,69 @@ export type Config = {
 };
 
 export interface MainnetContracts {
-  [core: string]: ethers.Contract;
-  tribe: ethers.Contract;
-  fei: ethers.Contract;
-  uniswapPCVDeposit: ethers.Contract;
+  core: Core;
```
Comment (Joeysantoro) [inline types/types.ts:140]: why remove [core: string]? I remember needing that for something

File: types/types.ts [hunk 2]
```diff
@@ -98,69 +137,69 @@ export type Config = {
...
-  rariPool8FeiPCVDeposit: ethers.Contract;
+  compositeOracle: CompositeOracle;
+  ethReserveStabilizer: EthReserveStabilizer;
+  ratioPCVController: RatioPCVController;
+  tribeReserveStabilizer: TribeReserveStabilizer;
+  feiRewardsDistributor: IFeiRewardsDistributor;
+  timelock: Timelock;
+  feiEthPair: IUniswapV2Pair;
+  rariPool8FeiPCVDeposit: ERC20CompoundPCVDeposit;
   rariPool8EthPCVDeposit: ethers.Contract;
```
Comment (Joeysantoro) [inline types/types.ts:156]: this an EthCompoundPCVDeposit

File: types/types.ts [hunk 3]
```diff
@@ -98,69 +137,69 @@ export type Config = {
...
-  uniswapPCVDeposit: ethers.Contract;
+  core: Core;
+  tribe: Tribe;
+  fei: Fei;
+  uniswapPCVDeposit: UniswapPCVDeposit;
   uniswapPCVController: ethers.Contract;
-  bondingCurve: ethers.Contract;
+  bondingCurve: BondingCurve;
   chainlinkEthUsdOracle: ethers.Contract;
   chainlinkFeiEthOracle: ethers.Contract;
```
Comment (Joeysantoro) [inline types/types.ts:147]: ChainlinkOracleWrapper

File: types/types.ts [hunk 4]
```diff
@@ -98,69 +137,69 @@ export type Config = {
...
 export interface MainnetContracts {
-  [core: string]: ethers.Contract;
-  tribe: ethers.Contract;
-  fei: ethers.Contract;
-  uniswapPCVDeposit: ethers.Contract;
+  core: Core;
+  tribe: Tribe;
+  fei: Fei;
+  uniswapPCVDeposit: UniswapPCVDeposit;
   uniswapPCVController: ethers.Contract;
```
Comment (Joeysantoro) [inline types/types.ts:144]: probably can remove

File: types/types.ts [hunk 5]
```diff
@@ -98,69 +137,69 @@ export type Config = {
...
+  curve3Metapool: IERC20;
+  erc20Dripper: ERC20Dripper;
+  tribalChiefOptimisticTimelock: Timelock;
+  staticPcvDepositWrapper: StaticPCVDepositWrapper;
+  collateralizationOracle: CollateralizationOracle;
+  collateralizationOracleWrapper: CollateralizationOracleWrapper;
+  collateralizationOracleKeeper: CollateralizationOracleKeeper;
+  tribeReserveStabilizerAddress: TribeReserveStabilizer;
+  pcvEquityMinter: PCVEquityMinter;
   tribeSplitter: ethers.Contract;
```
Comment (Joeysantoro) [inline types/types.ts:195]: This is an ERC20Splitter

File: types/types.ts [hunk 6]
```diff
@@ -98,69 +137,69 @@ export type Config = {
...
+  bentoBox: IMasterContractManager;
+  aaveEthPCVDripController: PCVDripController;
+  governorAlpha: GovernorAlpha;
+  tribalChief: TribalChief;
+  stakingTokenWrapper: StakingTokenWrapper;
+  feiTribePair: IUniswapV2Pair;
+  rariPool8Tribe: CErc20Delegator;
+  curve3Metapool: IERC20;
+  erc20Dripper: ERC20Dripper;
+  tribalChiefOptimisticTimelock: Timelock;
```
Comment (Joeysantoro) [inline types/types.ts:188]: OptimisticTimelock

**Review by Joeysantoro — DISMISSED** [review]
lgtm with some nits

File: types/types.ts (hunk 1 above)
Comment (ditchfieldcaleb) [reply types/types.ts:140]: Doesn't make any sense to me to have an array of cores' in this, and it doesn't seem to be breaking anything.

**Pull request merged**